Common/Securities/CashAmount.cs

```diff
@@ -47,5 +47,42 @@ public CashAmount(decimal amount, string currency)
             Amount = amount;
             Currency = currency;
         }
+
+        /// <summary>
+        /// Will determine if two <see cref="CashAmount"/> instances are equal
+        /// Useful to compare against the default instance
+        /// </summary>
+        /// <returns>True if <see cref="Currency"/> and <see cref="Amount"/> are equal</returns>
+        public static bool operator ==(CashAmount lhs, CashAmount rhs)
+        {
+            return Equals(lhs, rhs);
+        }
+
+        /// <summary>
+        /// Will determine if two <see cref="CashAmount"/> instances are different
+        /// Useful to compare against the default instance
+        /// </summary>
+        /// <returns>True if <see cref="Currency"/> or <see cref="Amount"/> are different</returns>
+        public static bool operator !=(CashAmount lhs, CashAmount rhs)
+        {
+            return !Equals(lhs, rhs);
+        }
+
+        /// <summary>
+        /// Used to compare two <see cref="CashAmount"/> instances.
+        /// Useful to compare against the default instance
+        /// </summary>
+        /// <param name="obj">The other object to compare with</param>
+        /// <returns>True if <see cref="Currency"/> and <see cref="Amount"/> are equal</returns>
+        public override bool Equals(object obj)
```

Comment:
Recommend also implementing `GetHashCode` - - in general, when overriding `Equals` on an immutable type you also want to override `GetHashCode`. This is especially true for struct types.

Also, ReSharper has a helper that generates the equality member functions by placing a cursor on the type, going to the ReSharper menu, then Edit, then Generate Code and selecting equality members from the pop-up. It will create a great hash function for you that guarantees a decent distribution and will also create the idiomatic equals which includes implementing `IEquatable<T>`.

Reply:
Nice, thanks for the tip!